test/datatable.jl

```diff
@@ -342,6 +342,40 @@ module TestDataTable
     @test find(c -> isa(c, NullableCategoricalArray), categorical!(DataTable(A=1:3, B=4:6), [1]).columns) == [1]
     @test find(c -> isa(c, NullableCategoricalArray), categorical!(DataTable(A=1:3, B=4:6), 1).columns) == [1]
 
+    @testset "unstack nullable promotion" begin
+        long = DataTable(Any[repeat([:c, :d], inner=12),
+                             vcat(linspace(0, 1, 12), linspace(2, 1, 12)),
+                             repeat(1:12, outer=2), repmat(repeat(1:3, inner=4), 2),
+                             repmat(repeat(1:4, inner=3), 2)],
+                         [:variable, :value, :id, :a, :b])
+        udt = unstack(long, :variable, :value)
+        @test udt == DataTable(Any[collect(1:12), repeat(1:3, inner = 4),
+                                   repeat(1:4, inner = 3),
+                                   Nullable{Float64}.(linspace(0, 1, 12)),
```

Comment:
Just `Nullable.` is enough. Or even `NullableArray` since that's a more common structure to use.